views/js/admin/src/components/Modal.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <div
+    :id="modalId"
+    class="fade modal"
+    tabindex="-1"
+    role="dialog">
+    <div
+      class="modal-dialog"
+      role="document">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h4
+            v-t="title"
+            class="modal-title" />
+          <button
+            type="button"
+            class="close"
+            data-dismiss="modal">
+            <MaterialIcon icon="close" />
+          </button>
+        </div>
+        <div
+          v-if="shown"
+          class="modal-body">
+          <div class="position-relative">
+            <LoaderOverlay v-show="loading" />
+          </div>
+          <slot
+            :modal-data="modalData"
+            :context="contextData" />
+        </div>
+        <div class="modal-footer">
+          <PsButton
+            class="btn-lg btn-outline-secondary"
+            label="cancel"
+            @click="() => onButtonClick('leave')" />
+          <PsButton
+            class="btn-lg btn-primary"
+            :label="saveLabel || 'save'"
+            data-type="save"
+            @click="() => onButtonClick('save')" />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { ModalCallbackProps, useModalContext } from '@/composables/context/useModalContext';
+import { PropType, computed, defineComponent, ref } from '@vue/composition-api';
+import LoaderOverlay from '@/components/common/LoaderOverlay.vue';
+import MaterialIcon from '@/components/common/MaterialIcon.vue';
+import PsButton from '@/components/common/PsButton.vue';
+import { contextProps } from '@/composables/props/contextProps';
+import { useContext } from '@/composables/context/useContext';
+
+export default defineComponent({
+  name: 'Modal',
+  components: {
+    LoaderOverlay,
+    MaterialIcon,
+    PsButton,
+  },
+
+  props: {
+    id: {
+      type: String,
+      default: null,
+    },
+
+    /**
+     * Callback to change behavior of save button. Note: You need to manually close the modal when using this.
+     */
+    onSave: {
+      type: Function as PropType<ModalCallbackProps['onSave']>,
+      default: null,
+    },
+
+    /**
+     * Callback to change behavior of leave button. Note: You need to manually close the modal when using this.
+     */
+    onLeave: {
+      type: Function as PropType<ModalCallbackProps['onLeave']>,
+      default: null,
+    },
+
+    title: {
+      type: String,
+      required: true,
+    },
+
+    saveLabel: {
+      type: String,
+      default: null,
+    },
+
+    ...contextProps,
```

Comment:
ik denk dat het beter is om dit bij de setup te plaatsen. ik vind props echt puur alleen maar bestemd voor waarden vanuit de parent.